### PR TITLE
s390x: ensure SecureExecution is enabled before sd-boot generation

### PR DESCRIPTION
**Pull request by nikita-dubrovskii**
By design `SecureExecution` is enabled during CoreOS build. So if systems runs in non-SE mode, skip `sd-boot` generation. 

https://github.com/coreos/fedora-coreos-tracker/issues/1205

### CODE REVIEW

**Review by cgwalters — APPROVED** [review]

File: src/libostree/ostree-bootloader-zipl.c [hunk 1]
```diff
@@ -109,6 +110,14 @@ _ostree_bootloader_zipl_write_config (OstreeBootloader  *bootloader,
   return TRUE;
 }
 
+static gboolean _ostree_secure_execution_is_enabled (GCancellable *cancellable) {
+  gsize len = 0;
+  g_autofree char *data = glnx_file_get_contents_utf8_at (-1, SECURE_EXECUTION_SYSFS_FLAG, &len, cancellable, NULL);
+  if ( !data )
```
Comment (cgwalters) [inline src/libostree/ostree-bootloader-zipl.c:116]: Can be fixed in a followup, but GNU C style has no spaces here, i.e. `if (!data)`

**Comment by dustymabe** [comment]
Looks like he addressed the code review comments. 👍

**Review by dustymabe — APPROVED** [review]

File: src/libostree/ostree-bootloader-zipl.c [hunk 2]
```diff
@@ -109,6 +110,14 @@ _ostree_bootloader_zipl_write_config (OstreeBootloader  *bootloader,
   return TRUE;
 }
 
+static gboolean _ostree_secure_execution_is_enabled (GCancellable *cancellable) {
```
Comment (jlebon) [inline src/libostree/ostree-bootloader-zipl.c:113]: I'm surprised at this signature (no `GError`). We need to make really sure here that we discern between "Secure Execution is disabled" (OK) and "couldn't read Secure Execution enablement" (hard error).

Is `/sys/firmware/uv/prot_virt_guest` always expected to exist on s390x machines, even if Secure Execution is disabled?

Comment (nikita-dubrovskii) [reply src/libostree/ostree-bootloader-zipl.c:113]: no GError, becase `/sys/firmware/uv/prot_virt_guest` exists only on machines with enabled SecureExecution. so if `openat` fails, error isn't propagated to the caller

Comment (jlebon) [reply src/libostree/ostree-bootloader-zipl.c:113]: Then ideally, to be strict we should handle `ENOENT` gracefully here but otherwise error out.

Comment (nikita-dubrovskii) [reply src/libostree/ostree-bootloader-zipl.c:113]: https://github.com/ostreedev/ostree/pull/2615

**Pull request merged**